server/conference-data/index.js

```diff
@@ -2,13 +2,35 @@ import R from 'ramda';
 
 const groupPartnersByLevel = R.groupBy(p => p.level.toLowerCase());
 
+export const pickSpeakersFromTalks = (talks) => {
+  // Talks contain speakers, but there is a chance that
+  // same speaker would participate in multiple talks
+  // We need to make sure we get a list of unique speakers
+  const speakers = [];
+  if (talks) {
+    talks.forEach((talk) => {
+      const filteredSpeakers = talk.speakers.filter((speaker) => {
+        if (!speakers.find((s) => s.id === speaker.id)) {
+          return speaker;
+        }
+        return null;
+      });
+
+      speakers.push(...filteredSpeakers);
+    });
+  }
+  return speakers;
+};
```

Comment:
This `O(n^2)`, it could be made `O(n)` by using an structure with constant lookup time for the de-duplicating of speakers.

It's also pretty hard to read in this imperative style. I would re-write it something like this (pseudocode, will need to use ramdba):

```js
talks.flat_map(talk => talk.speakers).uniq()
```